### ISSUE TITLE
Repository size is very large

### ISSUE THREAD

**Issue by nkrusch** [csci-1301/csci-1301.github.io#93]
**Reference**

https://api.github.com/repos/csci-1301/csci-1301.github.io

**Describe the issue**

Currently `"size": 2945739` ~ 3GB (the maximum allowed is 10 GB).

> Maximum git repository size is 10GB
The total git repository size will be limited to 10GB. Warnings and error messages will be alerted to you as your repository size grows, to help ensure that you are aware of reaching or approaching the size limits. If the repository size exceeds the limit, git pushes will be rejected. [ref](https://support.backlog.com/hc/en-us/articles/360051883894-Git-size-limit)

Cloning the repository takes a long time, but the actual files in various directories are "not that large" (except video, pdfs). The git history of commits of these large-ish files contributes to this size, and the large release size (now 33 MB) may also be a factor.

[`git-sizer`](https://github.com/github/git-sizer) gives me the following result:

```
./git-sizer --verbose
Processing blobs: 1752                        
Processing trees: 2714                        
Processing commits: 931                        
Matching commits to trees: 931                        
Processing annotated tags: 1                        
Processing references: 11                        
| Name                         | Value     | Level of concern               |
| ---------------------------- | --------- | ------------------------------ |
| Overall repository size      |           |                                |
| * Commits                    |           |                                |
|   * Count                    |   931     |                                |
|   * Total size               |   432 KiB |                                |
| * Trees                      |           |                                |
|   * Count                    |  2.71 k   |                                |
|   * Total size               |  1.03 MiB |                                |
|   * Total tree entries       |  28.0 k   |                                |
| * Blobs                      |           |                                |
|   * Count                    |  1.75 k   |                                |
|   * Total size               |   128 MiB |                                |
| * Annotated tags             |           |                                |
|   * Count                    |     1     |                                |
| * References                 |           |                                |
|   * Count                    |    11     |                                |
|     * Branches               |     1     |                                |
|     * Tags                   |     4     |                                |
|     * Remote-tracking refs   |     6     |                                |
|                              |           |                                |
| Biggest objects              |           |                                |
| * Commits                    |           |                                |
|   * Maximum size         [1] |  1.42 KiB |                                |
|   * Maximum parents      [2] |     2     |                                |
| * Trees                      |           |                                |
|   * Maximum entries      [3] |    98     |                                |
| * Blobs                      |           |                                |
|   * Maximum size         [4] |  33.0 MiB | ***                            |
|                              |           |                                |
| History structure            |           |                                |
| * Maximum history depth      |   812     |                                |
| * Maximum tag depth      [5] |     1     |                                |
|                              |           |                                |
| Biggest checkouts            |           |                                |
| * Number of directories  [6] |   101     |                                |
| * Maximum path depth     [6] |     7     |                                |
| * Maximum path length    [6] |   100 B   | *                              |
| * Number of files        [3] |   265     |                                |
| * Total size of files    [3] |  70.3 MiB |                                |
| * Number of symlinks         |     0     |                                |
| * Number of submodules       |     0     |                                |

[1]  a376508d21634bd5f3a2bff68ef7e40f19cc0fea
[2]  42f748072c18ba900f88f3a7bfd57bed120f110a
[3]  a9e56723d99e54649c1fa8c56e58ff67fae9d59c (refs/remotes/origin/gh-pages^{tree})
[4]  b8d8887ca7b5be2cacd7534dcc6a719b518d0e45 (refs/remotes/origin/gh-pages:release.zip)
[5]  f28b11528544f44175a3b7a91c16bcf1497d8be7 (refs/tags/2.0.0)
[6]  f1132e8f6535282e1600c547bc40cfee14886db1 (refs/heads/main^{tree})
```

**Suggested solutions**

- applying the git ways of [reducing history size](https://stackoverflow.com/questions/2116778/reduce-git-repository-size)
- Using [Git LFS](https://git-lfs.github.com/)
- Moving the large uneditable files (video, PDF) to box or similar shared drive
- Reset (partially?) the git history to bring the size down: it is unclear to me how this will impact the releases history and assets, and record of contributor; these should be preserved

**Comment by aubertc** [comment]
**tl;dr**: make a fresh clone, or clone only partial history. 

---

I have tried various git commands:

```
git maintenance run --task gc
git filter-repo --analyze
git rev-list --objects --all |   git cat-file --batch-check='%(objecttype) %(objectname) %(objectsize) %(rest)' |   sed -n 's/^blob //p' |   sort --numeric-sort --key=2 |   cut -c 1-12,41- |   $(command -v gnumfmt || echo numfmt) --field=2 --to=iec-i --suffix=B --padding=7 --round=nearest 
```
(To list the larger files) and other various tricks.

We will potentially need to reset the history at some point, for now I have added a warning at 
https://github.com/csci-1301/csci-1301.github.io#cloning

According to https://api.github.com/repos/csci-1301/csci-1301.github.io , our current size is 57195, which is 57MiB.
The [zip file](https://github.com/csci-1301/csci-1301.github.io/archive/refs/heads/main.zip) is around 45MiB, so that seems consistent.

A "fresh" clone is about 142.3 MiB (149,259,759): so, delete your previous folder and clone again if you want to gain space.
